jffs2/phoenix-rtos/locks.c

```diff
@@ -17,88 +17,51 @@
 #include "locks.h"
 
 
-void up_read(struct rw_semaphore *sem)
-{
-	mutexLock(sem->lock);
-
-	if (sem->cnt < 0x7fffffff) sem->cnt++;
-
-	if (sem->wwait)
-		condSignal(sem->wcond);
-	else if (sem->rwait)
-		condSignal(sem->rcond);
-
-	mutexUnlock(sem->lock);
-}
+/* Note: temporary RW semaphore implementation as mutex */
+/* It doesn't allow for multiple concurrent readers but enables priority inheritance */
+/* Fixes issues with both readers and writers being starved due to priority inversion */
+/* To be repaced by RW semaphore implementation in kernel */
```

Comment:
It's more of a personal insight than as a merge blocker, comments like:
```
/*
 * Note: temporary RW semaphore implementation as mutex
 * It doesn't allow for multiple concurrent readers but enables priority inheritance
 * Fixes issues with both readers and writers being starved due to priority inversion
 * To be repaced by RW semaphore implementation in kernel
 */
```
looks much better.

Reply:
I've changed the comment style to the proposed one (it seems like it's preferred over the other). Thanks!